src/core/frame.c

```diff
@@ -568,6 +568,7 @@ void MVM_frame_invoke(MVMThreadContext *tc, MVMStaticFrame *static_frame,
         MVMRegister *env       = static_frame->body.static_env;
         MVMuint8    *flags     = static_frame->body.static_env_flags;
         MVMint64     numlex    = static_frame->body.num_lexicals;
+        MVMCode     *cref      = (MVMCode *)frame->code_ref;
```

Comment:
Note that `frame` is rooted below, and that looking this up from the `frame` every time makes sure that we always get the correct value. If it's lifted out, then it must be also `MVMROOT`ed below.

Reply:
Thanks for pointing that out. I had extracted that coderef because it seemed easier to read and write in it's shorter form, but didn't realize that side-effect at the time. I updated the branch to root that variable as well, but if you think it would be better put back as it was I can do so.